### DISCUSSION TITLE
Is this a server request timeout`

### DISCUSSION THREAD

**Discussion by mdbergmann** [joaotavora/eglot#684] [General]
Hello.

I'm often seeing those in the eglot log when I put the curser to a new position.
Assumingly I should get some hover information, but that is not shown.

```
[internal] (id:25) Thu Apr 29 09:19:16 2021:
(:timed-out :textDocument/documentHighlight :id 25 :params
            (:textDocument
             (:uri "file:///myscalafile.scala")
             :position
             (:line 165 :character 30)))
```

Is this a timeout for an operation within eglot? Or is this a timeout from a request to the LSP server?

**Comment by joaotavora** [comment]
The server apparently took too long to respond to this request by Eglot  Maybe it tought it wouldn't be worth it anyway since you changed the file in the meantime, or maybe it was just busy, or maybe something else.  To know whence the request came, search for `id:25` in that buffer.

**Reply by mdbergmann** [reply]
I've asked because I'm seeing this a lot with Metals Scala LS. Having quite some trouble with this setup.
Thanks.
